rpc/cloudagent/network/loadbalancer/moc_cloudagent_loadbalancer.proto

```diff
@@ -27,10 +27,39 @@ message LoadbalancerInboundNatRule {
 	Protocol protocol = 4;
 }
 
+message LoadbalancerOutboundNatRule {
+	string name = 1;
+	uint32 frontendPort = 2;
+	uint32 backendPort = 3;
+	Protocol protocol = 4;
+}
+
 message LoadBalancingRule {
 	uint32 frontendPort = 1;
 	uint32 backendPort = 2;
 	Protocol protocol = 3;
+	ProbeReference probeRef = 4; // Update based on chat with Nithin
+}
+
+message Probe {
+	uint32 intervalInSeconds = 1;
+	uint32 numberOfProbes = 2;
+	Protocol protocol = 3;
+	uint32 port = 4; 
+	repeated LoadBalancingRule loadbalancingrules = 5; //read-only
+}
+
+message FrontEndIpConfiguration {
+	string privateIPAddress = 1; // required if using subnet
+	IPAllocationMethod allocationMethod = 2; // required
+	string subnetRef = 3; // required if using privateIPAddress. Pass just the name or fully qualified ref? 
+	PublicIPAddressReference publicIPAddress = 4; // required if not using privateIP and a subnet. resourceRef to publicIPAddress
+	
+	repeated LoadbalancerInboundNatRule inboundNatRules = 5; //read-only
+	repeated LoadbalancerOutboundNatRule outboundNatRules = 6; //read-only
+	repeated LoadBalancingRule loadbalancingrules = 7; //read-only
+
+	LoadBalancerVipConfigurationState configurationState = 8; // Do we need to surface this? how is it different from Probes?
```

Comment:
>LoadBalancerVipConfigurationState configurationState = 8; // Do we need to surface this? how is it different from Probes? [](http://example.com/codeflow?start=1&length=121)

configuration state will be a feature we will need to expose later. Let's not add it now as we dont know how that will look like.